app/controllers/users/rooms_controller.rb

```diff
@@ -24,16 +24,19 @@ def index
     end
     # ログインユーザーが参加しているすべてのroomから、user_idがログインユーザーでないレコードを取得
     @another_entries = Entry.where(room_id: my_room_ids).where.not(user_id: current_user.id)
+    # 最後にやりとりしたメッセージが新しい順にソートする処理
+    last_messages = @another_entries.map { |another_entry| another_entry.include_message }
+    sorted_messages = last_messages.sort_by! { |a| a[:created_at] }.reverse
+    @sorted_entries = sorted_messages.map { |sorted_message| sorted_message.room.entries.partner_of(current_user) }
```

Comment:
モデルにscopeを用意すれば下記のように呼べないですかね？controllerでしか持っていないcurrent_userなどは引数として渡せばいいかと。
```suggestion
    @sorted_entries = @another_entries.sorted_entries(current_user)
```